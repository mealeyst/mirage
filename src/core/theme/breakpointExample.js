import styled from 'styled-components'

/** @component */
export default styled.div`
  width: 300px;
  height: 300px;
  ${media.phone`background-color: green;`}
  ${media.tablet`background-color: blue;`}
  ${media.desktop`background-color: red;`}
`
