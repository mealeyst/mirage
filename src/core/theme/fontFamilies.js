import { injectGlobal } from 'styled-components'

/**
 * Global Fonts
**/
injectGlobal`
@font-face {
  font-family: 'ff-din-web';
  font-weight: 400;

  src: url('https://res.cloudinary.com/roa-canon/raw/upload/v1481915056/web/fonts/DINOT.otf');
}

@font-face {
  font-family: 'ff-din-web';
  font-weight: 400;
  font-style: italic;

  src: url('https://res.cloudinary.com/roa-canon/raw/upload/v1481915056/web/fonts/DINOT-Italic.otf');
}

@font-face {
  font-family: 'ff-din-web';
  font-weight: 500;

  src: url('https://res.cloudinary.com/roa-canon/raw/upload/v1481917384/web/fonts/DINOT-Medium.otf');
}


@font-face {
  font-family: 'ff-din-web';
  font-weight: 500;
  font-style: italic;

  src: url('https://res.cloudinary.com/roa-canon/raw/upload/v1481917385/web/fonts/DINOT-MediumItalic.otf');
}

@font-face {
  font-family: 'ff-din-web';
  font-weight: 800;

  src: url('https://res.cloudinary.com/roa-canon/raw/upload/v1481915056/web/fonts/DINOT-Bold.otf');
}

@font-face {
  font-family: 'ff-din-web';
  font-weight: 800;
  font-style: italic;

  src: url('https://res.cloudinary.com/roa-canon/raw/upload/v1481915056/web/fonts/DINOT-BoldItalic.otf');
}

@font-face {
  font-family: 'din-cond';
  font-weight: 400;

  src: url('https://res.cloudinary.com/roa-canon/raw/upload/v1481917384/web/fonts/DINOT-CondMedium.otf');
}
`

const fontFamilies = {
  dinCondensed: `"din-cond", "din-condensed-web", Arial, sans-serif`,
  dinRegular: `"ff-din-web", Helvetica, Arial, sans-serif`,
  caslon: `"adobe-caslon-pro", Helvetica, Arial, serif`
}

export default {
  headerFont: fontFamilies.dinCondensed,
  primaryFont: fontFamilies.dinRegular,
  secondaryFont: fontFamilies.caslon
}
