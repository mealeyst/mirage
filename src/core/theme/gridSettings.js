export default {
  columns: {
    mobile: 4,
    tablet: 10,
    desktop: 12
  },
  margins: {
    mobile: '3%',
    tablet: '7%',
    desktop: '7%'
  },
  gutter: '10px'
}
