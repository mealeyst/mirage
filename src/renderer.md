```js
const data = require('./pages/home.json');
<Renderer item={data} />
```
