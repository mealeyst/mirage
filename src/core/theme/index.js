import React from 'react'
import { ThemeProvider } from 'styled-components'
import { Fonts } from './fontFamilies'
import theme, { BaseFontSize } from './theme'

const ROATheme = (props) => {
  return (
    <main>
      <BaseFontSize />
      <Fonts />
      <ThemeProvider theme={theme} {...props} />
    </main>
  )
}

export default ROATheme
export { theme, ROATheme }
