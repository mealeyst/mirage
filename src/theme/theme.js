import { colors } from './colors'
import fontFamilies from './fontFamilies'

export default {
  colors: colors,
  fonts: fontFamilies,
  fixedPosition: 'relative'
}
