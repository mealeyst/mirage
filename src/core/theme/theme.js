import { injectGlobal } from 'styled-components'
import { colors } from './colors'
import media, { breakpoints, breakpointsVerbose } from './mediaQueries'
import fontFamilies from './fontFamilies'
import gridSettings from './gridSettings'

const base = injectGlobal`
  html {
    font-size: 10px;
  }
`

export default {
  breakpoints: breakpoints,
  colors: colors,
  fonts: fontFamilies,
  grid: gridSettings,
  fixedPosition: 'relative',
  media: media,
  breakpointsVerbose: breakpointsVerbose,
  base: base
}
