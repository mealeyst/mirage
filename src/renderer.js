import React from 'react'
import PropTypes from 'prop-types'

import * as Core from './core'
import * as Components from './components'
import * as Modules from './modules'


const COMPONENT_RENDER_MAP = {
  Core: {
    BackgroundImage: (item) => {
      return (
        <Core.BackgroundImage {...item.props}>
          {item.children.map(child => Renderer({item: child}))}
        </Core.BackgroundImage>
      )
    },
    Caption: (item) =>
      <Core.Caption {...item.props}>
        {item.children}
      </Core.Caption>,
    H1: (item) =>
      <Core.H1 {...item.props}>
        {item.children}
      </Core.H1>,
    H2: (item) =>
      <Core.H2 {...item.props}>
        {item.children}
      </Core.H2>,
    H3: (item) =>
      <Core.H3 {...item.props}>
        {item.children}
      </Core.H3>,
    H4: (item) =>
      <Core.H4 {...item.props}>
        {item.children}
      </Core.H4>,
    H5: (item) =>
      <Core.H5 {...item.props}>
        {item.children}
      </Core.H5>,
    H6: (item) =>
      <Core.H6 {...item.props}>
        {item.children}
      </Core.H6>,
    Label: (item) =>
      <Core.Label {...item.props}>
        {item.children}
      </Core.Label>,
    P: (item) =>
      <Core.P {...item.props}>
        {item.children}
      </Core.P>,
    Grid: (item) =>
      <Core.Grid>
        {item.children.map(child => Renderer({item: child}))}
      </Core.Grid>,
    Logo: () => <Core.Logo />,
    Sizer: (item) =>
      <Core.Sizer {...item.props}>
        {item.children.map(child => Renderer({item: child}))}
      </Core.Sizer>
  },
  Components: {
    InformationalSection: (item) =>
      <Components.InformationalSection>
        {item.children.map(child => Renderer({item: child}))}
      </Components.InformationalSection>,
    BackgroundVideo: (item) =>
      <Components.BackgroundVideo {...item.props}>
        {item.children.map(child => Renderer({item: child}))}
      </Components.BackgroundVideo>
  },
  Modules: {
    Page: (item) =>
      <Modules.Page>
        {item.children.map(child => Renderer({item: child}))}
      </Modules.Page>
  }
}

const Renderer = ({item}) => {
  try {
    return COMPONENT_RENDER_MAP[item.sys.type][item.sys.component](item.data)
  } catch(err) {
    console.warn(`It appears that you are tying to render an element that doesn't exist in the COMPONENT_RENDER_MAP`)
  }
}

Renderer.propTypes = {
  item: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.array,
    PropTypes.object
  ])
}

export default Renderer
