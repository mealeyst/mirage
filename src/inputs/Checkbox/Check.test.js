import React from 'react'
import renderer from 'react-test-renderer';
import 'jest-styled-components'

import { theme } from '../../theme'
import StyledCheck from './Check.base'

describe('(Base Component) Check', () => {
  test('should render correctly', () => {
    const tree = renderer
   .create(<StyledCheck theme={theme} />)
   .toJSON()
   expect(tree).toMatchSnapshot()
  })
})
