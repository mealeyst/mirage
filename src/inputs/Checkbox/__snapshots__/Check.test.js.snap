// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Base Component) Check should render correctly 1`] = `
.c0 {
  fill: none;
  stroke-width: 20;
  stroke-linecap: round;
  stroke: #FFFFFF;
}

<polyline
  className="c0"
  points="38.75 98.75 77.75 134.75 151.25 55.25"
/>
`;
